services/docker/docker-date.service.js

```diff
@@ -0,0 +1,129 @@
+'use strict'
+
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService, NotFound } = require('..')
+const { formatRelativeDate } = require('../text-formatters')
+const {
+  buildDockerUrl,
+  getDockerHubUser,
+  getMultiPageData,
+} = require('./docker-helpers')
+
+const buildSchema = Joi.object({
+  count: nonNegativeInteger.required(),
+  results: Joi.array().items(
+    Joi.object({
+      last_updated: Joi.date().iso().required(),
+      name: Joi.string().required(),
+      images: Joi.array().items(
+        Joi.object({
+          digest: Joi.string(),
+          architecture: Joi.string().required(),
+        })
+      ),
+    })
+  ),
+}).required()
+
+const queryParamSchema = Joi.object({
+  arch: Joi.string()
+    // Valid architecture values: https://golang.org/doc/install/source#environment (GOARCH)
+    .valid(
+      'amd64',
+      'arm',
+      'arm64',
+      's390x',
+      '386',
+      'ppc64',
+      'ppc64le',
+      'wasm',
+      'mips',
+      'mipsle',
+      'mips64',
+      'mips64le'
+    )
+    .default('amd64'),
+}).required()
```

Comment:
It looks like both these schemas are exact copy & pastes from the version service. If we want to use the exact same validation in multiple places, can we move it to `docker-helpers.js` and have `docker-date.js` and `docker-version.js` both import it from there.

Lets work out what we need first though.. if we bin the arch param, that changes things..